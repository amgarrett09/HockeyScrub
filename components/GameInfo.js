import React from 'react';
import PropTypes from 'prop-types';
import teamAbbreviations from '../constants/constants';
import '../css/game-info.css';

const GameInfo = ({
  homeTeam, awayTeam, homeScore, awayScore, gameState,
}) => (
  <div className="game-info">
    <div className="game-info__scores">
      <div className="row">
        <div className="col s6" data-test="team-name">
          {teamAbbreviations[awayTeam] || awayTeam}
        </div>
        <div className="col s6" data-test="team-score">
          {awayScore}
        </div>
      </div>

      <div className="row">
        <div className="col s6" data-test="team-name">
          {teamAbbreviations[homeTeam] || awayTeam}
        </div>
        <div className="col s6" data-test="team-score">
          {homeScore}
        </div>
      </div>
    </div>

    <div className="game-info__game-state">
      {gameState}
    </div>
  </div>
);

GameInfo.propTypes = {
  homeTeam: PropTypes.string.isRequired,
  awayTeam: PropTypes.string.isRequired,
  homeScore: PropTypes.number.isRequired,
  awayScore: PropTypes.number.isRequired,
  gameState: PropTypes.string.isRequired,
};

export default GameInfo;
