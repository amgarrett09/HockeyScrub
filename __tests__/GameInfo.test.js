/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import GameInfo from '../components/GameInfo';

describe('GameInfo', () => {
  const component = (
    <GameInfo
      awayTeam="Detroit Red Wings"
      homeTeam="Buffalo Sabres"
      awayScore={5}
      homeScore={4}
      gameState="FINAL"
    />
  );
  it('should have two team names and two team scores', () => {
    const wrapper = shallow(component);
    const teamNames = wrapper.find('[data-test="team-name"]');
    const teamScores = wrapper.find('[data-test="team-score"]');

    expect(teamNames.length).toBe(2);
    expect(teamScores.length).toBe(2);
  });

  it('should not change unexpectedly', () => {
    const wrapper = renderer.create(component);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
