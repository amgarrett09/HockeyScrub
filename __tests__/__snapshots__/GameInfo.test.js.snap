// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo should not change unexpectedly 1`] = `
<div
  className="game-info"
>
  <div
    className="game-info__scores"
  >
    <div
      className="row"
    >
      <div
        className="col s6"
        data-test="team-name"
      >
        DET
      </div>
      <div
        className="col s6"
        data-test="team-score"
      >
        5
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col s6"
        data-test="team-name"
      >
        BUF
      </div>
      <div
        className="col s6"
        data-test="team-score"
      >
        4
      </div>
    </div>
  </div>
  <div
    className="game-info__game-state"
  >
    FINAL
  </div>
</div>
`;
